sdk/eventhub/Azure.Messaging.EventHubs/src/ExponentialRetry.cs

```diff
@@ -50,13 +50,53 @@ public sealed class ExponentialRetry : Retry
         }
 
         /// <summary>
-        ///   Creates a new copy of the current <see cref="Retry" />, cloning its attributes into a new instance.
+        ///   Determines whether the two <see cref="ExponentialRetry" /> instances are equal.
         /// </summary>
         ///
-        /// <returns>A new copy of <see cref="Retry" />.</returns>
+        /// <param name="left">The first instance to compare.</param>
+        /// <param name="right">The second instance to compare.</param>
         ///
-        internal override Retry Clone() =>
-            new ExponentialRetry(_minimumBackoff, _maximumBackoff, _maximumRetryCount);
+        /// <returns><c>true</c> if the specified instances are equal; otherwise, <c>false</c>.</returns>
+        ///
+        public static bool operator ==(ExponentialRetry left,
```

Comment:
What's the reason for having equality comparison on this type?

Reply:
There are equality checks done at lower levels for determining whether or not an AMQP link needs to be closed and recreated.   We haven't yet gotten that far down the rabbit hole.

Reply:
I wonder if internal checks warrant implementing public comparison members

Reply:
Not sure that I follow the concern.   At this point the implementation is done and tested;  if there's no strong justification to remove, then I'm not sure that I see a benefit.

Reply:
Having less public APIs means fewer things to support if API is not useful for customers we shouldn't have it.

Reply:
Fair enough, that is an excellent point.   Will adjust the implementation to internalize.